.github/workflows/test.yml

```
@@ -10,9 +10,81 @@ on:
     branches:
       - main
       - staging
+  workflow_dispatch:
 
 jobs:
-  test:
+  question-service-tests:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      
+      - name: Set up .env
+        env:
+          QUESTION_FIREBASE_CREDENTIAL_PATH: ${{ vars.QUESTION_SERVICE_FIREBASE_CREDENTIAL_PATH }}
+          JWT_SECRET: ${{ secrets.JWT_SECRET }}
+        run: |
+          cd ./apps/question-service
+          
+          echo "FIREBASE_CREDENTIAL_PATH=$QUESTION_FIREBASE_CREDENTIAL_PATH" >> .env
+          echo "JWT_SECRET=$JWT_SECRET" >> .env
+      
+      - name: Set up credentials
+        env:
+          QUESTION_FIREBASE_JSON: ${{ secrets.QUESTION_SERVICE_FIREBASE_CREDENTIAL }}
+          QUESTION_FIREBASE_CREDENTIAL_PATH: ${{ vars.QUESTION_SERVICE_FIREBASE_CREDENTIAL_PATH }}
+        run: |
+          cd ./apps/question-service
+          echo "$QUESTION_FIREBASE_JSON" > "./$QUESTION_FIREBASE_CREDENTIAL_PATH"
+
+      - name: Setup Go
+        uses: actions/setup-go@v5
+        with: 
+          go-version: '1.23.x'
+
+      - name: Install Go dependencies
+        run: |
+          cd ./apps/question-service
+          go mod tidy
+
+      - name: Install firebase tools
+        run: curl -sL firebase.tools | bash
+      
+      - name: Run Go tests with Firebase emulator
+        run: firebase emulators:exec --only firestore 'cd ./apps/question-service'
```

This command should also include running the tests:

```suggestion
        run: firebase emulators:exec --only firestore 'cd ./apps/question-service && go test ./...'
```

---

oops. Ok fixed it, thanks for spotting this